### ISSUE TITLE
Problem getting schema default for scalar type

### ISSUE THREAD

**Issue by danielgtaylor** [pb33f/libopenapi#15]
While writing some tests I ran across this strange issue where a scalar default value is never loaded properly into the low or high level models. I've traced it to [SetField(...)](https://github.com/pb33f/libopenapi/blob/main/datamodel/low/model_builder.go#L450) where the default case of doing nothing is triggered so the field remains `nil`. I haven't gone too deep on understanding that code for a fix and figure you probably have a ton more context than me. Note that if I switch this to an `object` and make the default e.g. `{"hello": "world"}` that does load properly. Example:

```go
package main

import (
	"fmt"

	"github.com/pb33f/libopenapi"
)

var data = `
openapi: "3.1"
paths:
  /fails:
    get:
      responses:
        "200":
          content:
            application/json:
              schema:
                type: number
                default: 5
`

func main() {
	doc, err := libopenapi.NewDocument([]byte(data))
	if err != nil {
		panic(err)
	}

	result, errs := doc.BuildV3Model()
	if len(errs) > 0 {
		panic(errs)
	}

	m := result.Model

	fmt.Println("Fails", m.Paths.PathItems["/fails"].Get.Responses.Codes["200"].Content["application/json"].Schema.Schema().Default)
}
```

Result:

```sh
$ go run ./bug
Fails <nil>
```

Part of https://github.com/danielgtaylor/restish/issues/115

**Comment by daveshanley** [comment]
Interesting! I am looking into this for you, I should be able to patch it - in the next day or two.

**Comment by daveshanley** [comment]
Hi Daniel, 

This should now be resolved. The issue was that `any` types were being checked for object status, but there was no action otherwise. This check has been removed, and all defaults (primitive or objects) should be picked up correctly.

Thank you for helping make this library better.

The fix is available from `v0.2.8`
